lib/semian/grpc.rb

```diff
@@ -84,25 +84,55 @@ def disabled?
     def request_response(*, **)
       return super if disabled?
 
-      acquire_semian_resource(adapter: :grpc, scope: :request_response) { super }
+      scope = :request_response
+      acquire_semian_resource(adapter: :grpc, scope: scope) do
+        result = super
+        handle_operation(result, scope)
+        result
+      end
     end
 
     def client_streamer(*, **)
       return super if disabled?
 
-      acquire_semian_resource(adapter: :grpc, scope: :client_streamer) { super }
+      scope = :client_streamer
+      acquire_semian_resource(adapter: :grpc, scope: scope) do
+        result = super
+        handle_operation(result, scope)
+        result
+      end
     end
 
     def server_streamer(*, **)
       return super if disabled?
 
-      acquire_semian_resource(adapter: :grpc, scope: :server_streamer) { super }
+      scope = :server_streamer
+      acquire_semian_resource(adapter: :grpc, scope: scope) do
+        result = super
+        handle_operation(result, scope)
+        result
+      end
     end
 
     def bidi_streamer(*, **)
       return super if disabled?
 
-      acquire_semian_resource(adapter: :grpc, scope: :bidi_streamer) { super }
+      scope =:bidi_streamer
+      acquire_semian_resource(adapter: :grpc, scope: scope) do
+        result = super
+        handle_operation(result, scope)
+        result
+      end
+    end
+
+    def handle_operation(result, scope)
```

Comment:
CI now pass
https://github.com/jychen7/semian/actions/runs/2524645306

Reply:
> Also do you know what can we do with duplication lines

as `acquire_semian_resource` is defined at [Adapter](https://github.com/Shopify/semian/blob/v0.12.0/lib/semian/adapter.rb#L32) and such `handle_operation` is only valid in gRPC, I think we can define a private method here in `lib/semian/grpc`, then use it.
e.g.
```
def bidi_streamer(*, **)
  return super if disabled?

  acquire_semian_resource_grpc(scope: :bidi_streamer) { super }
end

def acquire_semian_resource_grpc(scope:)
  acquire_semian_resource(adapter: :grpc, scope: scope) do
    result = yield
    handle_operation(result, scope) if result.is_a?(::GRPC::ActiveCall::Operation)
    result
  end
end
```